src/plugins/plugin-http.ts

```diff
@@ -65,21 +61,27 @@ function makeRequestTrace(request, api) {
       return request.apply(this, arguments);
     }
 
-    options = isString(options) ? url.parse(options) : merge({}, options);
-    options.headers = options.headers || {};
+    var uri;
```

Comment:
This can be
```typescript
const uri = extractUrl(options);
```
because `extractUrl()` is called whether `options` is a string or not, right?

Reply:
It's a small optimization but I split it so that we wouldn't reconstruct a `uri` from an `options` object if no span is being created.

Reply:
Makes sense!